app/models/provider_enrichment.rb

```diff
@@ -58,4 +58,10 @@ def has_been_published_before?
   def publish(current_user)
     update(status: 'published', last_published_at: Time.now.utc, updated_by_user_id: current_user.id)
   end
+
+  def accrediting_provider_enrichment(provider_code)
+    accrediting_provider_enrichments&.find do |enrichment|
+      enrichment['UcasProviderCode'] == provider_code
```

Comment:
Some `accrediting_provider_enrichments` don't include the key `UcasProviderCode` here:

```
[33] pry(main)> Provider.find(12233).latest_enrichment.accrediting_provider_enrichments
  Provider Load (40.8ms)  SELECT  "provider".* FROM "provider" WHERE "provider"."id" = $1 LIMIT $2  [["id", 12233], ["LIMIT", 1]]
  ProviderEnrichment Load (0.3ms)  SELECT  "provider_enrichment".* FROM "provider_enrichment" WHERE "provider_enrichment"."provider_id" = $1 LIMIT $2  [["provider_id", 12233], ["LIMIT", 1]]
=> [{"Description"=>
   "Essex and Thames Primary SCITT is one of the UK's most established providers, having sustained an excellent reputation for 20 years.  Working in partnership with Canterbury Christ Church University the courses offer excellent value for money and an opportunity to train with some of the very best in the sector."}]
```

This isn't the first one I've seen, either.

Reply:
test coverage added for this case

Reply:
data investigation https://trello.com/c/jlK06Apv/1912-investigate-bad-data-in-accrediting-provider-enrichments